hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c

```diff
@@ -2448,6 +2448,67 @@ int launch_container_as_user(const char *user, const char *app_id,
     return exit_code;
 }
 
+static int parse_int (const char *str, int *val)
+{
+  long num;
+  char *endptr;
+
+  errno = 0;
+  num = strtol(str, &endptr, 0);
+
+  if (errno)
```

Comment:
what is supposed to change errno from is init value of 0 ?

Reply:
strtol will change errno if there is an error. You don't usually need to reset errno before calling a libc function, but integer parsing function are an exception.

To quote https://man7.org/linux/man-pages/man3/strtol.3.html
> Since strtol() can legitimately return 0, LONG_MAX, or LONG_MIN
> (LLONG_MAX or LLONG_MIN for strtoll()) on both success and
> failure, the calling program should set [errno](https://man7.org/linux/man-pages/man3/errno.3.html) to 0 before the
> call, and then determine if an error occurred by checking whether
> [errno](https://man7.org/linux/man-pages/man3/errno.3.html) has a nonzero value after the call.